src/main/java/edu/cornell/kfs/sys/service/impl/ReportWriterTextServiceImpl.java

```diff
@@ -66,6 +75,91 @@ public void initialize() {
 		this.writeHeader(title);
 	}
 
+	@Override
+	public void initialize(String fullyQualifiedDataFileName) {
+	    String dataFileNamePrefix = obtainDataFileName(fullyQualifiedDataFileName);
+	    if (StringUtils.isNotBlank(fullyQualifiedDataFileName)
+	            && StringUtils.isNotBlank(dataFileNamePrefix)
+	            && fullFilePathLimitWouldNotBeExceeded(dataFileNamePrefix)) {
+	        //use data file name as part of this report file name
+	        configureFileNamePrefixForThisReport(dataFileNamePrefix);
+	        LOG.info("initialize, data file name configured into report prefix name: {}", fileNamePrefix);
+	    } //else - use spring configured default report file name
+	    initialize();
+	}
+	
+	/**
+	 * This method generates a file name prefix from the fully qualified file name attribute
+	 *     (i.e.   directoryPath/fileName.fileExtension)
+	 * This functionality was created for the CreateAccountingDocumentReportItem xmlFileName attribute but could
+	 * be utilized by any other batch job that desired the report output file contain the input data file name
+	 * as the report file prefix. The input parameter is assumed to contain the fully qualified directory path
+	 * as well as a file name and extension. This method strips off both the directory path and file extension
+	 * returning just the file name portion of the string.
+	 * When both a directory path and file extension are not detected, a zero length string is returned.
+	 *
+	 * Example:
+	 *  Input parameter : /infra/work/staging/fp/accountingXmlDocument/fp_ib_netsuite_20240229_050035.xml
+	 *  Return value    : fp_ib_netsuite_20240229_050035
+	 *  
+	 *  @param fullyQualifiedDataFileName
+	 */
+	private String obtainDataFileName(String fullyQualifiedDataFileName) {
+	    String onlyDataFileName = KFSConstants.EMPTY_STRING;
+	    
+	    if (StringUtils.contains(fullyQualifiedDataFileName, KFSConstants.DELIMITER)
+	            && (StringUtils.contains(fullyQualifiedDataFileName, File.separator))) {
+	        onlyDataFileName = removeFileExtension(fullyQualifiedDataFileName);
+	        
+	        if (StringUtils.isNotBlank(onlyDataFileName)) {
+	            onlyDataFileName = StringUtils.substringAfterLast(onlyDataFileName, File.separator);
+	        }
+	    }
+	    return onlyDataFileName;
+	}
+	
+	private String removeFileExtension(String fullyQualifiedDataFileName) {
```

Comment:
Could this method be replaced with a call to `StringUtils.substringBeforeLast()` (or have such a call be the body of this method)? That would make the file extension removal much simpler.

Reply:
Adjusted.